datafusion/functions/src/math/gcd.rs

```diff
@@ -88,14 +88,15 @@ fn gcd(args: &[ArrayRef]) -> Result<ArrayRef> {
 
 /// Computes greatest common divisor using Binary GCD algorithm.
 pub fn compute_gcd(x: i64, y: i64) -> i64 {
-    let mut a = x.wrapping_abs();
-    let mut b = y.wrapping_abs();
+    // because the input values are i64, casting these u64's back to i64 causes no overflow
+    let mut a = x.unsigned_abs();
+    let mut b = y.unsigned_abs();
 
     if a == 0 {
-        return b;
+        return b as i64;
```

Comment:
If we move `unsigned_abs();` below, we don't need to cast

Reply:
Ah yes, nice catch! Thanks!